ethereum/retesteth/src/main/java/org/hyperledger/besu/ethereum/retesteth/NoRewardProtocolScheduleWrapper.java

```diff
@@ -41,54 +35,63 @@ public class NoRewardProtocolScheduleWrapper implements ProtocolSchedule {
-        Wei.ZERO, // block reward
-        original.getMiningBeneficiaryCalculator(),
-        original.getPrecompileContractRegistry(),
-        original.isSkipZeroBlockRewards(),
-        original.getGasCalculator(),
-        original.getGasLimitCalculator(),
-        original.getFeeMarket(),
-        original.getBadBlocksManager(),
-        Optional.empty(),
-        original.getWithdrawalsValidator(),
-        original.getWithdrawalsProcessor(),
-        original.getDepositsValidator(),
-        original.isPoS());
+  public ProtocolSpec getByBlockHeader(final ProcessableBlockHeader blockHeader) {
+    return delegate.getByBlockHeader(blockHeader);
   }
 
+  // TODO SLD - do we need this or a getByBlockHeader equivalent? Could be why referenceTests are
+  // failing?
+  //  @Override
+  //  public ProtocolSpec getByBlockNumber(final long number) {
+  //    final ProtocolSpec original = delegate.getByBlockNumber(number);
+  //    final BlockProcessor noRewardBlockProcessor =
+  //        new MainnetBlockProcessor(
+  //            original.getTransactionProcessor(),
+  //            original.getTransactionReceiptFactory(),
+  //            Wei.ZERO,
+  //            original.getMiningBeneficiaryCalculator(),
+  //            original.isSkipZeroBlockRewards(),
+  //            Optional.empty(),
+  //            delegate);
+  //    final BlockValidator noRewardBlockValidator =
+  //        new MainnetBlockValidator(
+  //            original.getBlockHeaderValidator(),
+  //            original.getBlockBodyValidator(),
+  //            noRewardBlockProcessor,
+  //            original.getBadBlocksManager());
+  //    final BlockImporter noRewardBlockImporter = new
+  // MainnetBlockImporter(noRewardBlockValidator);
+  //    return new ProtocolSpec(
+  //        original.getName(),
+  //        original.getEvm(),
+  //        original.getTransactionValidator(),
+  //        original.getTransactionProcessor(),
+  //        original.getPrivateTransactionProcessor(),
+  //        original.getBlockHeaderValidator(),
+  //        original.getOmmerHeaderValidator(),
+  //        original.getBlockBodyValidator(),
+  //        noRewardBlockProcessor,
+  //        noRewardBlockImporter,
+  //        noRewardBlockValidator,
+  //        original.getBlockHeaderFunctions(),
+  //        original.getTransactionReceiptFactory(),
+  //        original.getDifficultyCalculator(),
+  //        Wei.ZERO, // block reward
+  //        original.getMiningBeneficiaryCalculator(),
+  //        original.getPrecompileContractRegistry(),
+  //        original.isSkipZeroBlockRewards(),
+  //        original.getGasCalculator(),
+  //        original.getGasLimitCalculator(),
+  //        original.getFeeMarket(),
+  //        original.getBadBlocksManager(),
+  //        Optional.empty(),
+  //        original.getWithdrawalsValidator(),
+  //        original.getWithdrawalsProcessor(),
+  //        original.getDepositsValidator(),
+  //        original.isPoS());
+
+  //  }
```

Comment:
Do we need this or a getByBlockHeader equivalent? Could be why referenceTests are failing?

Reply:
@shemnon I've essentially migrated getBlockByNumber to getBlockByHeader. 
Do you know if the logic that I've commented out in here is actually used?

It's not covered by tests AFAICT, but I think it might be used if retesteth is run with certain inputs.

Reply:
I think I can just copy this code into the getByBlockHeader method above then. I think the break has already occurred during this commit actually, since that implementation of getBlockByNumber is no longer used: https://github.com/hyperledger/besu/commit/62f4a51191de52261acf099c3d78a898e12f240e

Reply:
IIRC this was the beyoncé rule one 😉 😆

Reply:
I've just migrated it here, hope this makes sense! https://github.com/hyperledger/besu/pull/5310/commits/b459bc31c8a9f1ef8dab1e232549d1a42f47ba8e

Reply:
what's the beyoncé rule??

Reply:
“If you liked it, then you shoulda put a test on it.”
https://abseil.io/resources/swe-book/html/ch11.html#:~:text=The%20Beyonc%C3%A9%20Rule&text=The%20straightforward%20answer%20is%3A%20test,an%20automated%20test%20for%20it.